src/unix/aix.c

```diff
@@ -881,24 +886,94 @@ void uv__fs_event_close(uv_fs_event_t* handle) {
 
 
 char** uv_setup_args(int argc, char** argv) {
-  return argv;
+  char** new_argv;
+  size_t size;
+  char* s;
+  int i;
+
+  if (argc <= 0)
+    return argv;
+
+  /* Save the original pointer to argv.
+   * AIX uses argv to read the process name.
+   * (Not the memory pointed to by argv[0..n] as on Linux.)
+   */
+  process_argv = argv;
+  process_argc = argc;
+
+  /* Calculate how much memory we need for the argv strings. */
+  size = 0;
+  for (i = 0; i < argc; i++)
+    size += strlen(argv[i]) + 1;
+
+  /* Add space for the argv pointers. */
+  size += (argc + 1) * sizeof(char*);
+
+  new_argv = uv__malloc(size);
+  if (new_argv == NULL)
+    return argv;
+  args_mem = new_argv;
+
+  /* Copy over the strings and set up the pointer table. */
+  s = (char*) &new_argv[argc + 1];
+  for (i = 0; i < argc; i++) {
+    size = strlen(argv[i]) + 1;
+    memcpy(s, argv[i], size);
+    new_argv[i] = s;
+    s += size;
+  }
+  new_argv[i] = NULL;
+
+  return new_argv;
 }
 
 
 int uv_set_process_title(const char* title) {
+  /* If this is the first time this is set,
+   * don't free and set argv[1] to NULL.
+   */
+  size_t len = strlen(title);
+  if( process_title_ptr != NULL )
+    uv__free(process_title_ptr);
+
+  /* We cannot free this pointer when libuv shuts down,
+   * the process may still be using it.
+   */
+  process_title_ptr = (char*) uv__malloc(len + 1);
+  if( process_title_ptr == NULL )
+    return 0;
```

Comment:
There's no actual error return for uv_set_process_title specified all implementations return 0 whether it works or is even supported by the platform. There's only return codes documented for uv_get_process_title not uv_set_process_title.
That said uv_set_process_title was declared to return `int` rather than `void` so I can if you like, I was just copying the current impls. Should I?